src/EnergyPlus/AirflowNetwork/src/Solver.cpp

```diff
@@ -12128,6 +12191,726 @@ namespace AirflowNetwork {
+                        MaxRough = DisSysCompDuctData(TypeNum).roughness;
+                        DynamicLoss = DisSysCompDuctData(TypeNum).TurDynCoef;
+                        DuctSizingSTFlag = true;
+                    }
+                } else {
+                    int NodeNum1 = AirflowNetworkLinkageData(AFNLinkNum).NodeNums[1];
+                    int CompNum1;
+                    iComponentTypeNum CompTypeNum1;
+                    CompNum1 = AirflowNetworkLinkageData(AFNLinkNum).CompNum;
+                    CompTypeNum1 = AirflowNetworkCompData(CompNum1).CompTypeNum;
+                    if (CompTypeNum1 == iComponentTypeNum::DWC) {
+                        AirflowNetworkLinkageData(AFNLinkNum).DuctLineType = iDuctLineType::SupplyTrunk;
+                        TypeNum = AirflowNetworkCompData(CompNum1).TypeNum;
+                        SumLength += DisSysCompDuctData(TypeNum).L;
+                        MaxRough = max(MaxRough, DisSysCompDuctData(TypeNum).roughness);
+                        DynamicLoss += DisSysCompDuctData(TypeNum).TurDynCoef;
+                        // NodeNum1 = AirflowNetworkLinkageData(AFNLinkNum).NodeNums[0];
+                        DuctSizingSBFlag = true;
+                    }
+                    while (NodeNum1 != NodeSplitter) {
+                        for (AFNLinkNum1 = 1; AFNLinkNum1 <= AirflowNetworkNumOfLinks; AFNLinkNum1++) {
+                            if (NodeNum1 == AirflowNetworkLinkageData(AFNLinkNum1).NodeNums[0]) {
+                                CompNum1 = AirflowNetworkLinkageData(AFNLinkNum1).CompNum;
+                                CompTypeNum1 = AirflowNetworkCompData(CompNum1).CompTypeNum;
+                                if (CompTypeNum1 == iComponentTypeNum::DWC) {
+                                    AirflowNetworkLinkageData(AFNLinkNum1).DuctLineType = iDuctLineType::SupplyTrunk;
+                                    TypeNum = AirflowNetworkCompData(CompNum1).TypeNum;
+                                    SumLength += DisSysCompDuctData(TypeNum).L;
+                                    MaxRough = max(MaxRough, DisSysCompDuctData(TypeNum).roughness);
+                                    DynamicLoss += DisSysCompDuctData(TypeNum).TurDynCoef;
+                                    NodeNum1 = AirflowNetworkLinkageData(AFNLinkNum1).NodeNums[1];
+                                    DuctSizingSTFlag = true;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+                if (DuctSizingSTFlag) {
+                    Real64 Velocity = 0.0;
+                    Real64 flowrate = DisSysCompCVFData(1).FlowRate / m_state.dataEnvrn->StdRhoAir;
+                    if (AirflowNetworkSimu.ductSizeMethod == AirflowNetworkSimuProp::DuctSizeMethod::MaxVelocity) {
+                        SupplyTrunkD = sqrt(4.0 * flowrate / AirflowNetworkSimu.DuctSizeMaxV / 3.1415926);
```

Comment:
Do we not have a `constexpr Real64 Pi` somewhere?

Reply:
Hopefully not, but mathematical constants have long been a C/C++ issue because.... Microsoft I think? Anyway, in C++20 there is a [`numbers` ](https://en.cppreference.com/w/cpp/header/numbers) header that will solve all of our problems. It unfortunately does not include `answer_to_the_ultimate_question of_life_the_universe_and_everything`, so we're still on our own for the existential stuff.

Reply:
What I meant was Pi is already defined in `DataGlobalConstants.hh`. You can just write `DataGlobalConstants::Pi` (or hopefully in the future `Const::Pi`) instead of 3.1415926. Unless of course, you want to impress people with your knowledge of Pi to seven decimal places ... in which case you would still be wrong because I believe you would need to round up to 3.1415927. 😄

Reply:
I am going to use DataGlobalConstants::Pi.